kornia/contrib/kmeans.py

```diff
@@ -0,0 +1,155 @@
+# based on https://github.com/subhadarship/kmeans_pytorch
+
+import torch
+
+
+class KMeans:
+    def __init__(
+        self,
+        num_clusters: int,
+        cluster_centers: torch.Tensor,
+        tolerance: float = 10e-4,
+        max_iterations: int = 0,
+        device: torch.device = torch.device('cpu'),
+        seed: int = None,
+    ) -> None:
+        if num_clusters == 0:
+            raise ValueError("num_clusters can't be 0")
+
+        self.num_clusters = num_clusters
+        self.cluster_centers = cluster_centers
+        self.tolerance = tolerance
+        self.max_iterations = max_iterations
+        self.device = device
+
+        self.final_cluster_assignments = None
+        self.final_cluster_centers = None
+
+        if seed is not None:
+            torch.manual_seed(seed)
+
+    def get_cluster_centers(self) -> torch.Tensor:
+        if self.final_cluster_centers is None:
+            raise ValueError("Model has not been fit to a dataset")
+        return self.final_cluster_centers.cpu()
+
+    def get_cluster_assignments(self) -> torch.Tensor:
+        if self.final_cluster_assignments is None:
+            raise ValueError("Model has not been fit to a dataset")
+        return self.final_cluster_assignments.cpu()
+
+    def _initialise_cluster_centers(self, X: torch.Tensor, num_clusters: int) -> torch.Tensor:
+        """Chooses num_cluster points from X as the initial cluster centers.
+
+        Args:
+            X: 2D input tensor to be clustered
+            num_clusters: number of desired cluster centers
+        Return:
+            2D Tensor with num_cluster rows
+        """
+        num_samples = len(X)
+        perm = torch.randperm(num_samples)
+        idx = perm[:num_clusters]
+        initial_state = X[idx]
+        return initial_state
+
+    def _pairwise_euclidean_distance(self, data1: torch.Tensor, data2: torch.Tensor) -> torch.Tensor:
+        """Computes pairwise distance between 2 sets of vectors.
+
+        Args:
+            data1: 2D tensor of shape N, D
+            data2: 2D tensor of shape C, D
+        Return:
+            2D tensor of shape N, C
+        """
+        # N*1*D
+        A = data1.unsqueeze(dim=1)
+        # 1*C*D
+        B = data2.unsqueeze(dim=0)
+        distance = (A - B) ** 2.0
+        # return N*C matrix for pairwise distance
+        distance = distance.sum(dim=-1)
+        return distance
+
+    def fit(self, X: torch.Tensor) -> None:
+        """Iterative KMeans clustering till a threshold for shift in cluster centers
```

Comment:
Do string don’t follow conventions

Reply:
I tried searching for docstring conventions. Not sure what changes are needed. Tried running docformatter, nothing changed. Please let me know.